Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,336 @@
+            {
+                playerList.Add(player);
+            }
+            else
+                continue;
+
+            var pref = (HumanoidCharacterProfile) _prefs.GetPreferences(player.UserId).SelectedCharacter;
+            if (pref.AntagPreferences.Contains(comp.HeadRevPrototypeId))
+                prefList.Add(player);
+        }
+        if (playerList.Count == 0)
+            return;
+
+        _sawmill.Error("Math");
+        var headRevs = Math.Clamp(allPlayers.Count / HeadRevsPerPlayer, 1, MaxRevHeads);
+        for (var revs = 0; revs < headRevs; revs++)
+        {
+            IPlayerSession headRev;
+            if (prefList.Count == 0)
+            {
+                if (playerList.Count == 0)
+                {
+                    break;
+                }
+                headRev = _random.PickAndTake(playerList);
+            }
+            else
+            {
+                headRev = _random.PickAndTake(prefList);
+                playerList.Remove(headRev);
+            }
+            var mind = headRev.Data.ContentData()?.Mind;
+            if (mind == null)
+            {
+                continue;
+            }
+            _mindSystem.AddRole(mind, new RevolutionaryRole(mind, _prototypeManager.Index<AntagPrototype>(comp.HeadRevPrototypeId)));
+            var inCharacterName = string.Empty;
+            if (mind.OwnedEntity != null)
+            {
+                AddComp<RevolutionaryRuleComponent>(mind.OwnedEntity.Value);
+                AddComp<HeadRevolutionaryComponent>(mind.OwnedEntity.Value);
+                AddComp<RevolutionaryComponent>(mind.OwnedEntity.Value);
+                _stationSpawningSystem.EquipStartingGear(mind.OwnedEntity.Value, _prototypeManager.Index<StartingGearPrototype>(comp.HeadRevGearPrototypeId), null);
+                _npcFaction.RemoveFaction(mind.OwnedEntity.Value, "NanoTrasen", false);
+                _npcFaction.AddFaction(mind.OwnedEntity.Value, "Revolutionary");
+                inCharacterName = MetaData(mind.OwnedEntity.Value).EntityName;
+                comp.HeadRevs.Add(inCharacterName, headRev.Name);
+            }
+            if (mind.Session != null)
+            {
+                var message = Loc.GetString("head-rev-role-greeting");
+                var wrappedMessage = Loc.GetString("chat-manager-server-wrap-message", ("message", message));
+                _chatManager.ChatMessageToOne(Shared.Chat.ChatChannel.Server, message, wrappedMessage, default, false, mind.Session.ConnectedClient, Color.MediumBlue);
+            }
+        }
+        AssignCommandStaff(cantBeAntag);
+    }
+
+    //Assigns command staff based on their weight (real). I THINK all head weight is above 10 so uh let me know if not.
+    //I wanted to do sec as well but I'm not sure if they have a specific weight and they have glasses so if they get converted it's their own fault.
```

Comment:
This should likely be a list of jobs considered heads on the revolutionary rule. Not the cleanest, but less likely to break in unusual ways (it'll break predictably instead).

Reply:
I think I did it but the way might be kind of scuffed.